### ISSUE TITLE
For loops generate error in loss function

### ISSUE THREAD

**Issue by sinestic** [VectXmy/FCOS.Pytorch#5]
Hi! great job!
I have noticed that there is a useless loop in computed losses.
For example, here a snippet from compute_cls_loss
``` python
for pred in preds:
        pred=pred.permute(0,2,3,1)
        pred=torch.reshape(pred,[batch_size,-1,class_num])
        preds_reshape.append(pred)
preds=torch.cat(preds_reshape,dim=1)#[batch_size,sum(_h*_w),class_num]
```
the permute generate an error because there is a shift in 4 dims but pred have only 3 dims. Reshape in batch_size is useless because you are looping in batch_size dim.
Same situation in compute_cnt_loss and compute_reg_loss

Can i suggest to delete the for loop and use only the permute and reshape methods? 

``` python
pred=pred.permute(0,2,3,1)
pred=torch.reshape(pred,[batch_size,-1,class_num])
```
like the snippet below.

Thank you!

**Comment by VectXmy** [comment]
pred has 4 dims(batchsize,channels,h,w). preds is List, containing all branch pred. Here we want to merge all pred while each pred has different size in dim3 and dim4. So the loop is necessary. Thank you all the same!

**Comment by sinestic** [comment]
ok, but the permute still generate an error
